typegraph/typegraph/providers/prisma/runtimes/prisma.py

```diff
@@ -304,42 +308,150 @@ def executeRaw(self, query: str, *, effect: Effect) -> t.func:
     def find_unique(self, tpe: Union[t.struct, t.NodeProxy]) -> t.func:
         self.__manage(tpe)
         typegen = self.__typegen
+        _pref = get_name_generator("Unique", tpe)
+        # output = typegen.get_out_type(base_output)
+        output = typegen.add_nested_count(tpe)
         return t.func(
             t.struct(
-                {"where": typegen.get_where_type(tpe).named(f"{tpe.name}WhereUnique")}
+                {
+                    # findUnique's where expression
+                    # does not allow AND, OR, NOT
+                    "where": typegen.gen_query_where_expr(
+                        tpe, exclude_extra_fields=True
+                    )
+                    .named(_pref("Where"))
+                    .optional()
+                }
             ),
-            typegen.get_out_type(tpe).named(f"{tpe.name}UniqueOutput").optional(),
+            output.named(_pref("Output")).optional(),
             PrismaOperationMat(self, tpe.name, "findUnique", effect=effects.none()),
         )
 
     def find_many(self, tpe: Union[t.struct, t.NodeProxy]) -> t.func:
         self.__manage(tpe)
         typegen = self.__typegen
+        _pref = get_name_generator("Many", tpe)
+        rel_cols = tpe.props.keys()
+        # output = typegen.get_out_type(base_output)
+        output = typegen.add_nested_count(tpe)
         return t.func(
             t.struct(
                 {
-                    "where": typegen.get_where_type(tpe)
-                    .named(f"{tpe.name}Where")
-                    .optional()
+                    "where": typegen.gen_query_where_expr(tpe)
+                    .named(_pref("Where"))
+                    .optional(),
+                    "orderBy": typegen.get_order_by_type(tpe)
+                    .named(_pref("OrderBy"))
+                    .optional(),
+                    "take": t.integer().named(_pref("Take")).optional(),
+                    "skip": t.integer().named(_pref("Skip")).optional(),
+                    "distinct": t.array(t.enum(rel_cols))
+                    .named(_pref("Distinct"))
+                    .optional(),
                 }
             ),
-            t.array(typegen.get_out_type(tpe).named(f"{tpe.name}Output")),
+            t.array(output.named(_pref("Output"))),
             PrismaOperationMat(self, tpe.name, "findMany", effect=effects.none()),
         )
 
+    def aggregate(self, tpe: Union[t.struct, t.NodeProxy]) -> t.func:
+        self.__manage(tpe)
+        typegen = self.__typegen
+        _pref = get_name_generator("Aggregate", tpe)
+        tpe_nums = typegen.extract_number_types(tpe)
+        return t.func(
+            t.struct(
+                {
```

Comment:
I would suggest to add any arg type to the parameter of the function so user can override them if they want:

```
    def aggregate(self, tpe: Union[t.struct, t.NodeProxy], where=typegen.gen_query_where_expr(tpe)
                     .named(_pref("Where"))) -> t.func:
```

would allow 

```
db.aggregate(mytype, where=typegen.gen_query_where_expr(mytype, id=t.string().set(XXXX)))
```